extensions/amp-story/1.0/amp-story.js

```diff
@@ -1041,21 +1038,25 @@ export class AmpStory extends AMP.BaseElement {
    * @private
    */
   getInitialPageId_(firstPageEl) {
-    const isActualPage = pageId =>
-      findIndex(this.pages_, page => page.element.id === pageId) >= 0;
     const historyPage = /** @type {string} */ (getHistoryState(
       this.win,
       HistoryState.PAGE_ID
     ));
 
     if (isExperimentOn(this.win, 'amp-story-branching')) {
       const maybePageId = parseQueryString(this.win.location.hash)['page'];
-      if (maybePageId && isActualPage(maybePageId)) {
+      const maybePage = maybePageId
+        ? this.element.querySelector(`#${escapeCssSelectorIdent(maybePageId)}`)
+        : null;
+      if (maybePageId && maybePage) {
         return maybePageId;
       }
     }
 
-    if (historyPage && isActualPage(historyPage)) {
+    const maybeHistoryPage = historyPage
+      ? this.element.querySelector(`#${escapeCssSelectorIdent(historyPage)}`)
+      : null;
+    if (historyPage && maybeHistoryPage) {
```

Comment:
Not sure why you removed the function that factorized that code, we could simply update the `isActualPage` function and keep the current implementation
```
const isActualPage = pageId => !!this.element.querySelector(`#${escapeCssSelectorIdent(historyPage)}`);
```

Reply:
Yep my bad. Done.